src/hotspot/cpu/riscv/gc/g1/g1_riscv.ad

```diff
@@ -162,11 +162,11 @@ instruct g1EncodePAndStoreN(indirect mem, iRegP src, iRegPNoSp tmp1, iRegPNoSp t
   ins_pipe(istore_reg_mem);
 %}
 
-instruct g1CompareAndExchangeP(iRegPNoSp res, indirect mem, iRegP oldval, iRegP newval, iRegPNoSp tmp1, iRegPNoSp tmp2)
+instruct g1CompareAndExchangeP(iRegPNoSp res, indirect mem, iRegP oldval, iRegP newval, iRegPNoSp tmp1, iRegPNoSp tmp2, rFlagsReg cr)
 %{
   predicate(UseG1GC && n->as_LoadStore()->barrier_data() != 0);
   match(Set res (CompareAndExchangeP mem (Binary oldval newval)));
-  effect(TEMP res, TEMP tmp1, TEMP tmp2);
```

Comment:
should the effect of `res` be changed from `TEMP` to `DEF_TEMP`?

Reply:
Ah, that looks like a refactoring change to me. Maybe go with a separate change if it's necessary? Seems not strongly related to this PR.

Reply:
Yes, make sense to me.